src/H5Ppublic.h

```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *
+ *          The parameters to the callback function are defined as follows:
+ *
+ *          <table>
+ *           <tr>
+ *            <td>\Code{const char * name}</td>
+ *            <td>IN: The name of the property being copied</td>
+ *           </tr>
+ *           <tr>
+ *            <td>\Code{size_t size}</td>
+ *            <td>IN: The size of the property in bytes</td>
+ *           </tr>
+ *           <tr>
+ *            <td>\Code{void * value}</td>
+ *            <td>IN/OUT: The value for the property being copied</td>
+ *           </tr>
+ *          </table>
+ *
+ *          The \p prp_copy routine may modify the value to be set and those
+ *          changes will be stored as the new value of the property. If
+ *          the \p prp_copy routine returns a negative value, the new
+ *          property value is not copied into the property and the \p prp_copy
+ *          routine returns an error value.
+ *
+ *          The \p prp_close routine is called when a property list with this
+ *          property is being closed. The #H5P_prp_close_func_t callback
+ *          function is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb1_t_snip
+ *
+ *          The parameters to the callback function are defined as follows:
+ *
+ *          <table>
+ *           <tr>
+ *            <td>\Code{const char * name}</td>
+ *            <td>IN: The name of the property in the list</td>
+ *           </tr>
+ *           <tr>
+ *            <td>\Code{size_t size}</td>
+ *            <td>IN: The size of the property in bytes</td>
+ *           </tr>
+ *           <tr>
+ *            <td>\Code{void * value}</td>
+ *            <td>IN: The value for the property being closed</td>
+ *           </tr>
+ *          </table>
+ *
+ *          The \p prp_close routine may modify the value passed in, but the
+ *          value is not used by the library when the \p prp_close routine returns.
+ *          If the \p prp_close routine returns a negative value, the property
+ *          list close routine returns an error value but the property list is
+ *          still closed.
+ *
+ * \note \Bold{Programming Note for C++ Developers Using C Functions:}
+ * If a C routine that takes a function pointer as an argument is called
+ * from within C++ code, the C routine should be returned from normally.
+ * Examples of this kind of routine include callbacks such as
+ * H5P_SET_ELINK_CB and H5P_SET_TYPE_CONV_CB and functions such as
+ * H5T_CONVERT and H5E_WALK2.
+ * Exiting the routine in its normal fashion allows the HDF5 C library to
+ * clean up its work properly. In other words, if the C++ application jumps
+ * out of the routine back to the C++ “catch” statement, the library is not
+ * given the opportunity to close any temporary data structures that were
+ * set up when the routine was called. The C++ application should save some
+ * state as the routine is started so that any problem that occurs might be
+ * diagnosed.
```

Comment:
Let's not repeat this block literally everywhere! Would you mind appending the following single line to `aliases`?
```
ALIASES += cpp_c_api_note="\note \Bold{C++ Developers using HDF5 C-API functions beware:}\nIf a C routine that takes a function pointer as an argument is called from within C++ code, the C routine should be returned from normally. Examples of this kind of routine include callbacks such as H5Pset_elink_cb() and H5Pset_type_conv_cb() and functions such as H5Tconvert() and H5Ewalk2().\n Exiting the routine in its normal fashion allows the HDF5 C library to clean up its work properly. In other words, if the C++ application jumps out of the routine back to the C++ \c catch statement, the library is not given the opportunity to close any temporary data structures that were set up when the routine was called. The C++ application should save some state as the routine is started so that any problem that occurs might be diagnosed."
```
Then you can replace this paragraph by just `\cpp_c_api_note`